fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreOpeningTest.java

```diff
@@ -118,6 +119,7 @@ public void open() throws Exception {
             RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecords1Proto.getDescriptor());
 
             FDBRecordStore recordStore = storeBuilder(context, metaDataBuilder).createOrOpen();
+            assertTrue(recordStore.getVersionChanged());
```

Comment:
Maybe I'm missing it, but is there a test for what happens if you call `checkVersion` twice, and the version is changed during the first call but then not in the second? I think you were concerned an update could be lost in that case. (It's also possible that that's already covered in this test.)

Reply:
added this.